lib/new_relic/agent/span_event_primitive.rb

```diff
@@ -122,6 +123,10 @@ def for_datastore_segment(segment) # rubocop:disable Metrics/AbcSize
           agent_attributes[DB_STATEMENT_KEY] = truncate(segment.nosql_statement, DB_STATEMENT_MAX_BYTES)
         end
 
+        if segment.params[:backtrace]
```

Comment:
the backtrace is only ever added to the params hash when the duration is exceeded. That check is already done by the time we get here, so no need to check the duration against the threshold again